src/main/1.3/scala/com/holdenkarau/spark/testing/DataframeGenerator.scala

```diff
@@ -0,0 +1,49 @@
+package com.holdenkarau.spark.testing
+
+import org.apache.spark.SparkContext
+import org.apache.spark.rdd.RDD
+import org.apache.spark.sql.{Row, DataFrame}
+import org.apache.spark.sql.types._
+import org.scalacheck.{Arbitrary, Gen}
+
+object DataframeGenerator {
+
+  /**
+    * Generates a DataFrame for the required Schama
+    *
+    * @param sc Spark Context
+    * @param schema The required Schema
+    * @param minPartitions defaults to 1
+    *
+    * @return
+    */
+  def genDataFrame(sc: SparkContext, schema: StructType, minPartitions: Int = 1): Arbitrary[DataFrame] = {
```

Comment:
I'd change this to take in a SQLContext rather than a SparkContext since if the user is testing with some hive based transformations it might be useful to have them pass in their existing context.